plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/views/VulnerabilitiesView.java

```diff
@@ -189,6 +191,18 @@ public void onFilterLoad(Servers retrievedServers, Applications retrievedApplica
 		}
 	};
 
+	private SeverityFilterListener severityFilterListener = new SeverityFilterListener() {
+		@Override
+		public void onSeverityFilterLoad(EnumSet<RuleSeverity> severities) {
+			if (!severities.isEmpty()) {
+				currentTraceFilterForm.setSeverities(severities);
+			} else {
+				currentTraceFilterForm.setSeverities(null);
+			}
+			startRefreshJob();

```

Comment:
I think that the page offset should be reset to first page, because if you are in the last page, you change the filters and get less vulnerabilities you will apparently be sent to first page without visible vulnerabilities and will have to move the pager again to be able to actually see the filtered vulnerabilities.

Reply:
@rduran0 Fixed